operator/pkg/operator.go

```diff
@@ -174,13 +235,44 @@ func (o *Operator) Start(ctx context.Context) error {
 				o.Logger.Fatal("Could not subscribe to new tasks V3")
 			}
 		case newBatchLogV2 := <-o.NewTaskCreatedChanV2:
-			go o.handleNewBatchLogV2(newBatchLogV2)
+			go o.handleNewBatchLogV2(newBatchLogV2, batchProcessorChan)
 		case newBatchLogV3 := <-o.NewTaskCreatedChanV3:
-			go o.handleNewBatchLogV3(newBatchLogV3)
+			go o.handleNewBatchLogV3(newBatchLogV3, batchProcessorChan)
```

Comment:
Instead of passing the `batchProcessedChan` as parameter, it would be best for the relevant functions to call something like `o.registerBatchProcessed(int)` which sends a message to the channel. 
Hint: `type Operator struct {` should probably have a chan for this to work.